src/domain/model.ts

```diff
@@ -20,10 +20,6 @@ export type AdoptablePets<Location, Pets> = {
   readonly pets: Pets
 }
 
-export type GetPets<Location, Pets> = {
-  getPets(l: Location): Promise<Pets>
-}
```

Comment:
What other things might an object called `GetPets` do?

Reply:
I'm thinking about general patterns that scale.  I love function types, but you might also want an interface with several cohesive operations.  For example, imagine you also want to enable adding new pets.  It might be reasonable to have a Pets interface with getPets and addPet operations.  Grouping them into a single interface makes sense to me in cases like that.

So, I had been thinking that instead of making developers think about whether to pick a function type or object type, maybe it makes sense just to always use an object type, since that'll scale to multi-method interfaces.

Reply:
Ah I see what you mean.  

This is starting to feel a lot like a case-by-case basis set of decisions :D 

I predict a side effect of the object type will be a lot of interfaces that look an awful lot like an ORM or a "REST-y/CRUD"  type thing focused on manipulating a single domain entity.  Which will result in questions like, is it `Pets.addOwner` or `Owner.addPet`?  Obviously the interfaces don't have to follow an entity, but I find this kind of thinking _really_ hard to break away from when the code so easily supports it.  

For me, using function types encourages "data-in" -> "data-out" thinking that isn't necessarily tied to an object model.  `adopt(owner, pet)`.  

I'm not sure either way is inherently better, but as a default I've been enjoying the latter lately